src/material/timepicker/timepicker-input.ts

```diff
@@ -0,0 +1,421 @@
+   */
+  registerOnChange(fn: (value: any) => void): void {
+    this._onChange = fn;
+  }
+
+  /**
+   * Implemented as a part of `ControlValueAccessor`.
+   * @docs-private
+   */
+  registerOnTouched(fn: () => void): void {
+    this._onTouched = fn;
+  }
+
+  /**
+   * Implemented as a part of `ControlValueAccessor`.
+   * @docs-private
+   */
+  setDisabledState(isDisabled: boolean): void {
+    this._accessorDisabled.set(isDisabled);
+  }
+
+  /**
+   * Implemented as a part of `Validator`.
+   * @docs-private
+   */
+  validate(control: AbstractControl): ValidationErrors | null {
+    return this._validator(control);
+  }
+
+  /**
+   * Implemented as a part of `Validator`.
+   * @docs-private
+   */
+  registerOnValidatorChange(fn: () => void): void {
+    this._validatorOnChange = fn;
+  }
+
+  /** Gets the element to which the timepicker popup should be attached. */
+  getOverlayOrigin(): ElementRef<HTMLElement> {
+    return this._formField?.getConnectedOverlayOrigin() || this._elementRef;
+  }
+
+  /** Focuses the input. */
+  focus(): void {
+    this._elementRef.nativeElement.focus();
+  }
+
+  ngOnDestroy(): void {
+    this.getOverlayOrigin().nativeElement.removeEventListener('click', this._handleClick);
+    this._timepickerSubscription?.unsubscribe();
+    this._localeSubscription.unsubscribe();
+  }
+
+  /** Gets the ID of the input's label. */
+  _getLabelId(): string | null {
+    return this._formField?.getLabelId() || null;
+  }
+
+  /** Handles clicks on the input or the containing form field. */
+  private _handleClick = (): void => {
+    this.timepicker().open();
+  };
+
+  /** Handles the `input` event. */
+  protected _handleInput(value: string) {
+    const currentValue = this.value();
+    const date = this._dateAdapter.parseTime(value, this._dateFormats.parse.timeInput);
+    const hasChanged = !this._dateAdapter.sameTime(date, currentValue);
+
+    if (!date || hasChanged || !!(value && !currentValue)) {
+      // We need to fire the CVA change event for all nulls, otherwise the validators won't run.
+      this._assignUserSelection(date, true);
+    } else {
+      // Call the validator even if the value hasn't changed since
+      // some fields change depending on what the user has entered.
+      this._validatorOnChange?.();
+    }
+  }
+
+  /** Handles the `blur` event. */
+  protected _handleBlur() {
+    const value = this.value();
+
+    // Only reformat on blur so the value doesn't change while the user is interacting.
+    if (value && this._isValid(value)) {
+      this._formatValue(value);
+    }
+
+    this._onTouched?.();
+  }
+
+  /** Handles the `keydown` event. */
+  protected _handleKeydown(event: KeyboardEvent) {
+    // All keyboard events while open are handled through the timepicker.
+    if (this.timepicker().isOpen()) {
+      return;
+    }
+
+    if (event.keyCode === ESCAPE && !hasModifierKey(event) && this.value() !== null) {
+      event.preventDefault();
+      this.value.set(null);
```

Comment:
This part seems surprising to me. Why does ESCAPE clear the input? The datepicker doesnt do this

Reply:
I got this off [the W3C example](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-autocomplete-list/): 

>  Escape 
> If the listbox is displayed, closes it
> If the listbox is not displayed, clears the textbox.